dali/kernels/tensor_view.h

```diff
@@ -414,6 +414,32 @@ TensorListView<StorageGPU, T, ndim> make_tensor_list_gpu(T *data, TensorListShap
   return { data, std::move(shape) };
 }
 
+
+/**
+ * @brief Get a subtensor by slicing along outermost dimension at index `idx`
+ * 
+ * @details Produces tensor, for which number of dimensions is reduced by 1.
+ * Removed dimension is outer-most (e.g. for shape {3,2,4,6} produces {2,4,6}).
+ * Data inside the tensor is extracted according to provided index.
+ * Data is not copied.
+ *
+ * Example:
+ * tv.data = [[1, 2, 3], [4, 5, 6]]       (shape: [2, 3])
+ * oust_dimension(tv, 1) -> [4, 5, 6]     (shape: [3])
+ *
+ * @param source Source TensorView
+ * @param idx Index inside dimension, along which data is extracted
+ * @return TensorView with reduced dimensionality
+ */
+template<typename StorageBackend, typename DataType, int ndims>
```

Comment:
```suggestion
template<int out_ndim_ = InferDimensions, typename StorageBackend, typename DataType, int ndim,
    int out_ndim = (out_ndim_ == InferDimensions ? ndim == DynamicDimensions ? DynamicDimensions : ndim-1 : out_ndim_>
```
This allows user to make a slice with known static number of dimensions from a tensor with dynamic number of dimensions.